packages/box/src/Box.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { HTMLElementProps } from '@leafygreen-ui/lib';
+import omit from 'lodash/omit';
+
+type DivProps<T> = HTMLElementProps<'div'> &
+  T & {
+    href?: undefined;
```

Comment:
I think you can remove `{ href?: undefined }` because this should only be the `div` props and the shared props. Presumably if the shared props include `href` or `component`, their union should be treated as `AnchorProps` or `CustomElementProps` anyway.

Reply:
Removing this line causes Typescript to complain in the `usesAnchorElement` function, because it believes that `href` should not be present in `BoxProps`. I think since we do a loose check of the value of `href` (null/undefined before determining whether or not the component should use AnchorProps) it's okay that we specifically say that `href` should not be defined in div props.

Reply:
If so, I think then we should probably add `component?: undefined` as well, to keep things consistent. Does that work? If not, no worries, I don't want this to block a merge.

Reply:
Nvm, fixed this!